Backend/lib/payment/src/Lib/Payment/Storage/Queries/PaymentOrder.hs

```diff
@@ -11,57 +11,74 @@
 import Kernel.Prelude
-import Kernel.Storage.Esqueleto as Esq
 import Kernel.Types.Id
-import Kernel.Utils.Common (getCurrentTime)
+import Kernel.Utils.Common (MonadFlow, getCurrentTime)
 import qualified Lib.Payment.Domain.Types.PaymentOrder as DOrder
-import Lib.Payment.Storage.Tabular.PaymentOrder
-
-findById :: Transactionable m => Id DOrder.PaymentOrder -> m (Maybe DOrder.PaymentOrder)
-findById = Esq.findById
-
-findByShortId :: Transactionable m => ShortId DOrder.PaymentOrder -> m (Maybe DOrder.PaymentOrder)
-findByShortId shortId =
-  findOne $ do
-    order <- from $ table @PaymentOrderT
-    where_ $ order ^. PaymentOrderShortId ==. val (getShortId shortId)
-    return order
-
-findLatestByPersonId :: Transactionable m => Text -> m (Maybe DOrder.PaymentOrder)
-findLatestByPersonId personId =
-  findOne $ do
-    order <- from $ table @PaymentOrderT
-    where_ $ order ^. PaymentOrderPersonId ==. val personId
-    orderBy [desc $ order ^. PaymentOrderCreatedAt]
-    limit 1
-    return order
-
-create :: DOrder.PaymentOrder -> SqlDB ()
-create = Esq.create
-
-updateStatus :: DOrder.PaymentOrder -> SqlDB ()
+import qualified Lib.Payment.Storage.Beam.PaymentOrder as BeamPO
+import qualified Sequelize as Se
+
+findById :: MonadFlow m => Id DOrder.PaymentOrder -> m (Maybe DOrder.PaymentOrder)
+findById (Id paymentOrder) = findOneWithKV [Se.Is BeamPO.id $ Se.Eq paymentOrder]
+
+findByShortId :: MonadFlow m => ShortId DOrder.PaymentOrder -> m (Maybe DOrder.PaymentOrder)
+findByShortId (ShortId shortId) = findAllWithOptionsKV [Se.Is BeamPO.shortId $ Se.Eq shortId] (Se.Desc $ BeamPO.createdAt) (Just 1) Nothing <&> listToMaybe
+
+findLatestByPersonId :: MonadFlow m => Text -> m (Maybe DOrder.PaymentOrder)
+findLatestByPersonId personId = findAllWithOptionsKV [Se.Is BeamPO.personId $ Se.Eq personId] (Se.Desc $ BeamPO.createdAt) (Just 1) Nothing <&> listToMaybe
+
+create :: MonadFlow m => DOrder.PaymentOrder -> m ()
+create = createWithKV
+
+updateStatus :: MonadFlow m => DOrder.PaymentOrder -> m ()
 updateStatus order = do
   now <- getCurrentTime
-  Esq.update $ \tbl -> do
-    set
-      tbl
-      [ PaymentOrderStatus =. val order.status,
-        PaymentOrderUpdatedAt =. val now
-      ]
-    where_ $ tbl ^. PaymentOrderId ==. val order.id.getId
-
-updateStatusToExpired :: Id DOrder.PaymentOrder -> SqlDB ()
-updateStatusToExpired orderId = do
+  updateWithKV [Se.Set BeamPO.status order.status, Se.Set BeamPO.updatedAt now] [Se.Is BeamPO.id $ Se.Eq $ getId order.id]
+
+updateStatusToExpired :: (MonadFlow m) => Id DOrder.PaymentOrder -> m ()
+updateStatusToExpired (Id orderId) = do
   now <- getCurrentTime
-  Esq.update $ \tbl -> do
-    set
-      tbl
-      [ PaymentOrderStatus =. val Payment.CLIENT_AUTH_TOKEN_EXPIRED,
-        PaymentOrderUpdatedAt =. val now
-      ]
-    where_ $ tbl ^. PaymentOrderId ==. val orderId.getId
+  updateWithKV [Se.Set BeamPO.updatedAt now] [Se.Is BeamPO.id $ Se.Eq orderId]
```

Comment:
This is also, incorrect. Lets recheck all queries.

Reply:
fixed it